nodejs/eks/authenticationMode.ts

```diff
@@ -64,6 +72,38 @@ export function validateAuthenticationMode(args: ClusterOptions) {
             }
         });
     }
+    return args;
+}
+
+// Validate promptly if possible, otherwise validate in the Promise chain underlying the Output and ensure that the
+// input is gated on the validation. Unfortunately since apply always unwraps, the validate function required here needs
+// to be able to handle both unwrapped and normal forms.
+function validatedInput<T>(
```

Comment:
This trick didn't work as neatly as I hoped. I hoped to make a `pulumi.Input<T>` that is gated on the validation. Unfortunately I couldn't wrestle the jest framework to be able to assert against failing pulumi.Output values :/ so simply coercing `pulumi.Input<T>` to `pulumi.Output<T>` does not cut it because I couldn't fix the unit tests. Instead this is further refined to handle immediately available T on the stack and only jump into an apply chain if that is not possible. Unfortunately since apply always unwraps and there doesn't seem to be a non-unwrapping variant available, the `validate` function  has this signature:

```
 validate: (value: T|pulumi.Unwrap<T>|undefined) => void
```

where the ideal signature would be:

```
 validate: (value: T|undefined) => void
```